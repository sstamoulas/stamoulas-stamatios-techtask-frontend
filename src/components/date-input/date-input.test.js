import React from 'react'
import { Provider } from 'react-redux'
import { act, create } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
 
import DateInput from './date-input.component'
import MealDateActionTypes from './../../redux/mealDate/mealDate.types'
 
const mockStore = configureStore([])
const mealDate = new Date('07/14/2020')

describe('Date Input Component', () => {
  let store
  let component
 
  beforeEach(() => {
    store = mockStore({
      mealDate: {
        mealDate,
      },
    })
 
    store.dispatch = jest.fn()
 
    component = create(
      <Provider store={store}>
        <DateInput />
      </Provider>
    )
  })
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should dispatch an action when the date changes', () => {
    act(() => {
      component.root.findByType('input').props.onChange({ target: { value: mealDate }})
    })
 
    expect(store.dispatch).toHaveBeenCalledTimes(1)
    expect(store.dispatch).toHaveBeenCalledWith({ 
      type: MealDateActionTypes.UPDATE_MEAL_DATE, 
      payload: mealDate 
    })
  })
})
