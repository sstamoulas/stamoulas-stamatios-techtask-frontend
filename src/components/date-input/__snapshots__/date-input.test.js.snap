// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Input Component should render with given state from Redux store 1`] = `
<div
  className="input-container"
>
  <label
    htmlFor="date"
  >
    Meal Date:  
  </label>
  <input
    id="date"
    onChange={[Function]}
    type="date"
    value="2020-07-14"
  />
</div>
`;
